elements/lisk-api-client/src/transaction.ts

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+
+import {
+	signTransaction,
+	signMultiSignatureTransaction,
+	computeMinFee,
+} from '@liskhq/lisk-transactions';
+import { getAddressAndPublicKeyFromPassphrase } from '@liskhq/lisk-cryptography';
+import {
+	decodeTransaction,
+	encodeTransaction,
+	getTransactionAssetSchema,
+	decodeAccount,
+} from './codec';
+import { Channel, RegisteredSchemas, NodeInfo } from './types';
+
+export interface CreateTransactionOptions {
+	nonce?: bigint;
+	includeSenderSignature?: boolean;
+	multisignatureKeys?: {
+		mandatoryKeys: Buffer[];
+		optionalKeys: Buffer[];
+	};
+}
+
+export interface MultiSignatureKeys {
+	readonly mandatoryKeys: Buffer[];
+	readonly optionalKeys: Buffer[];
+	readonly numberOfSignatures: number;
+}
+
+export class Transaction {
+	private readonly _channel: Channel;
+	private readonly _schema: RegisteredSchemas;
+	private readonly _nodeInfo: NodeInfo;
+
+	public constructor(channel: Channel, registeredSchema: RegisteredSchemas, nodeInfo: NodeInfo) {
+		this._channel = channel;
+		this._schema = registeredSchema;
+		this._nodeInfo = nodeInfo;
+	}
+
+	public async create(
+		input: {
+			moduleID?: number; // id takes priority
+			moduleName?: string;
+			assetID?: number; // id takes priority
+			assetName?: string;
+			fee: bigint;
+			nonce: bigint;
+			senderPublicKey: Buffer;
+			asset: Record<string, unknown>;
+			signatures?: Buffer[];
+		},
+		passphrase: string,
+		options?: {
+			nonce?: bigint;
+			includeSenderSignature?: boolean;
+			multisignatureKeys?: {
+				mandatoryKeys: Buffer[];
+				optionalKeys: Buffer[];
+			};
+		},
+	): Promise<Record<string, unknown>> {
+		const txInput = input;
+		const networkIdentifier = Buffer.from(this._nodeInfo.networkIdentifier, 'hex');
+		const { publicKey, address } = getAddressAndPublicKeyFromPassphrase(passphrase);
+		const accountHex = await this._channel.invoke<string>('app:getAccount', {
+			address: address.toString('hex'),
+		});
+		const account = decodeAccount(Buffer.from(accountHex, 'hex'), this._schema);
+		if (!txInput.moduleID) {
+			if (!txInput.moduleName) {
+				throw new Error('Missing moduleId and moduleName');
+			} else {
+				const registeredModule = this._nodeInfo.registeredModules.find(
+					module => module.name === input.moduleName,
+				);
+				txInput.moduleID = registeredModule?.id ? registeredModule.id : txInput.moduleID;
+			}
+		}
+		if (txInput.assetID === undefined || null) {
```

Comment:
```suggestion
		if (txInput.assetID === undefined || txInput.assetID === null) {
```
or u could check `typeof txInput.assetID !== 'number'`